apps/website/prisma/schema.prisma

```diff
@@ -241,4 +249,88 @@ model OutgoingWebhook {
 
   user          User?          @relation(fields: [userId], references: [id])
   giveawayEntry GiveawayEntry?
+
+  @@index([userId], name: "OutgoingWebhook_userId")
+}
+
+model LinkAttachment {
+  id              String @id @default(cuid())
+  type            String
+  name            String
+  title           String
+  alternativeText String
+  caption         String
+  description     String
```

Comment:
What is the difference between these three fields? The UI seems to only show alt text + caption, and I wonder if those should be combined?

Reply:
> One avenue for abuse I noticed -- if you add an image to a post and then remove it, it doesn't look like we delete it from S3? Meaning someone could sit on the submit page forever just adding images and removing them, using our S3 bucket as free storage? Equally, someone could add images in there and then close the tab, using our S3 bucket as free storage? Maybe we should only upload them to S3 when the post is actually submitted, or cleanup any images in the bucket that aren't tied to a post after a timeout (30 minutes maybe)?

Yep, thats on my todo list in #5 . That is also why I added the file storage object references to the database - to be able to clean them. One thing that hindered me from implementing this already is that I am unsure if its okay to use the newly introduced vercel cron jobs or if we would need another service (e.g. the long-running node server) for the cleanup task? 

If we were to upload all files on submit we would need to either orchestrate that in the client making the user wait or pass everything as multipart to the serverless function and relay the files to DO. I think its easier to upload them separately and clean up if they are not used.

For the possible abuse of the file storage until the file gets cleaned or approved I do not really have a solution. We can upload them without public access and only make them public after the post is submitted. We cant really wait for the approval though, because the reviewer would need to be able to see them to review them unless we use an api route to make  them viewable for authorized users.

This is all pretty new to me (that is serverless + s3), so feel free to suggest different approaches! :)